content/scripts/document-extractor.js

```diff
@@ -217,6 +217,60 @@ function _addSingleSectionBCD($) {
   if (data === undefined) {
     return [];
   }
+
+  // First extract a map of all release data, keyed by (normalized) browser
+  // name and the versions.
+  // You'll have a map that looks like this:
+  //
+  //   'chrome_android': {
+  //      '28': {
+  //        release_data: '2012-06-01',
+  //        release_notes: '...',
+  //        ...
+  //
+  // The reason we extract this to a locally scoped map, is so we can
+  // use it to augment the `__compat` blocks for the latest version
+  // when (if known) it was added.
+  const browserReleaseData = new Map();
+  Object.entries(browsers).forEach(([name, browser]) => {
+    const releaseData = new Map();
+    for (const [version, data] of Object.entries(browser.releases || [])) {
+      if (data) {
+        releaseData.set(version, data);
+      }
+    }
+    browserReleaseData.set(name, releaseData);
+  });
+
+  // We never need this data, after the release info has been extracted
+  // for each 'version_added'.
+  Object.values(browsers).forEach((browser) => {
+    // Remove because it's added weight which we don't need in the
+    // state data sent to the client eventually.
+    delete browser.releases;
```

Comment:
Makes me wonder, is there more things like this we can delete?

Reply:
But I'd like to save that for another day. This PR makes a *huge* difference to start with. Remember to refer to https://gist.github.com/peterbe/8410c625ae0fce515adea92284451ba1